src/components/FeatureViews/AdminPortal/PanelManager.js

```diff
@@ -0,0 +1,306 @@
+            }
+        });
+        return field;
+    });
+
+    const fieldsWithDefaults = addDefaults(props.fields);
+
+
+
+    const viewRecordRow = (record) => {
+        const recordElements = [];
+        
+        fieldsWithDefaults.forEach((field, index) => {
+            recordElements.push(
+
+                    <Grid item xs={field["col-width"]} md={field["col-width"]}>
+                        <Typography align={field["align"]}>
+                            {Object.values(record)[index + 1]}
+                        </Typography>
+                    </Grid>
+
+            );
+        })
+
+        return (
+            <Paper square={true} className={"category-row"} >
+                <Grid container alignItems={"center"}>
+                
+                    {recordElements}
+                    <Grid item xs={2} md={2}>
+                    <IconButton
+                        onClick={editRecord(record.id)}
+                        >
+                    <EditIcon/>
+                    </IconButton>
+                </Grid>
+                </Grid>
+
+            </Paper>
+        )
+
+};
+const editRecord = (id) => (e) => {
+    e.preventDefault();
+    console.log("it worked");
+    let editingRecord = recordList.find((record) => {return record.id === id});
+    let recordToUpload;
+    if(editingRecord) {
+        recordToUpload = {
+            id: editingRecord.id,
+            name: editingRecord.name,
+            description: editingRecord.description,
+        };
+
+        //! Should be updateRecord...ideally takes this as a prop
+        
+        props.updateFunction(id, recordToUpload);
+    }
+    editingRecord.editing = !editingRecord.editing;
+    let updatedRecordList = recordList.map((record) => {
+        if(record.id === id){
+            return editingRecord;
+        } else {
+            return record;
+        }
+    });
+    console.log(updatedRecordList);
+    setRecordList(updatedRecordList);
+};
+
+const handleEditRecord = (type, id) => (e) => {
+    let editingRecord = recordList.find((record) => {return record.id === id});
+    switch(type) {
+        case "name":
+            editingRecord.name = e.target.value;
+            break;
+        case "description":
+            editingRecord.description = e.target.value;
+            break;
+    }
+    
+
+    //! RecordToUpdateIndex = recordList.indexOf(editingRecord)
+    //! update record... => recordList[indexOf(editingRecord)] = editingRecord
+    let updatedRecordList = recordList.map((record) => {
+        if (record.id == id) {
+            return editingRecord;
+        } else {
+            return record;
+        }
+    });
+    setRecordList(updatedRecordList);
+};
+
+const editRecordRow = (record) => {
+    console.log("edit");
+    return (
+        <Paper square={true} className={"category-row"} >
+            <Grid container alignItems={"center"}>
+                <Grid item xs={3} md={3} >
```

Comment:
If you have these columns fixed, then we won't be able to add additional fields (since we can only have 2 fields where 1 field takes up three columns and the other takes seven columns). The number of columns should be determined by the user.